### ISSUE TITLE
Clarify test documentation

### ISSUE THREAD

**Issue by YanDoroshenko** [http4s/http4s#1704]
Sorry for being annoying, but I did not find any good testing documentation. I've gone through testing subproject and tests for various other subprojects, but was unable to find any comprehensive info.
An example like:
```
val service = ExistingServer.service
val request = Request("https://localhost", 443, , Method.POST, """{ "text" : "hello world" }""")

val response = service.respond(request)
response.status shouldBe 200
```
would be extremely useful, for I've been only able to find 1) a standalone client (and I don't understand, how do you run both server and client at the same time) 2) some kind of test with a manually created HTTP header & stuff.
Thanks in advance.

**Comment by jmcardon** [comment]
Hi, thanks for the interest to start off.

> I've gone through testing subproject and tests for various other subprojects, but was unable to find any comprehensive info.

It depends what sort of thing you're trying to test. Our services (as of 0.18) are simply an `Request[F] => OptionT[F, Response]`. That is, there's no necessarily specific testing framework you need to test such a thing. You only need a reference to the service, and you can run your service without a live server to test whether it functions properly.

[this](https://github.com/jmcardon/tsec/blob/master/tsec-http4s/src/test/scala/tsec/authentication/RequestAuthenticatorSpec.scala#L80) is an example of relatively raw testing without the http4s testing framework.

That said, someone other than me knows the framework testing helpers we put out, so there's that.

**Comment by YanDoroshenko** [comment]
@jmcardon Perfect, thank you.

**Comment by rossabaker** [comment]
A tut devoted to testing is in progress on #1718.  We'll close this with that.  If that doesn't tell us what we want to know, please comment there.

**Comment by jmcardon** [comment]
Can this be closed? #1718 was merged
